tools/scoring/tests/dimensions/entity_identification_test.py

```diff
@@ -17,34 +17,178 @@
+    self.assertEqual(none_score.incorrect_total(), 0)
+
+    # Inherited result properties. These are `None` by virtue of the ceiling
+    # being falsy: that is to say, there was nothing to score against.
+    self.assertEqual(none_score.result_all, None)
+    self.assertEqual(none_score.result_reporting, None)
+    self.assertEqual(none_score.result_virtual, None)
+
+  def testEntityIdentificationHighestScore(self):
+    """When correct==ceiling, the resulting score is 1.0."""
+    highest_score = EntityIdentification(
+        deserialized_files=self.highest_score_argument).evaluate()
+
+    # Directly assigned attributes
+    self.assertEqual(highest_score.correct_reporting, 1)
+    self.assertEqual(highest_score.correct_ceiling_reporting, 1)
+    self.assertEqual(highest_score.incorrect_reporting, 0)
+
+    self.assertEqual(highest_score.correct_virtual, 1)
+    self.assertEqual(highest_score.correct_ceiling_virtual, 1)
+    self.assertEqual(highest_score.incorrect_virtual, 0)
+
+    # Inherited calculated attributes
+    self.assertEqual(highest_score.correct_total(), 2)
+    self.assertEqual(highest_score.correct_ceiling(), 2)
+    self.assertEqual(highest_score.incorrect_total(), 0)
+
+    # Inherited result properties
+    self.assertEqual(highest_score.result_all, 1.0)
+    self.assertEqual(highest_score.result_reporting, 1.0)
+    self.assertEqual(highest_score.result_virtual, 1.0)
+
+  def testEntityIdentificationLowestScore(self):
+    """When correct==0, the resulting score is -1.0."""
+    lowest_score_argument = {
+        PROPOSED: self.none_score_argument[PROPOSED],
+        SOLUTION: self.highest_score_argument[SOLUTION]
+    }
+    lowest_score = EntityIdentification(
+        deserialized_files=lowest_score_argument).evaluate()
+    # Directly assigned attributes
+    self.assertEqual(lowest_score.correct_reporting, 0)
+    self.assertEqual(lowest_score.correct_ceiling_reporting, 1)
+    self.assertEqual(lowest_score.incorrect_reporting, 1)
+
+    self.assertEqual(lowest_score.correct_virtual, 0)
+    self.assertEqual(lowest_score.correct_ceiling_virtual, 1)
+    self.assertEqual(lowest_score.incorrect_virtual, 1)
+
+    # Inherited calculated attributes
+    self.assertEqual(lowest_score.correct_total(), 0)
+    self.assertEqual(lowest_score.correct_ceiling(), 2)
+    self.assertEqual(lowest_score.incorrect_total(), 2)
+
+    # Inherited result properties
+    self.assertEqual(lowest_score.result_all, -1.0)
+    self.assertEqual(lowest_score.result_reporting, -1.0)
+    self.assertEqual(lowest_score.result_virtual, -1.0)
+
+  def testEntityIdentificationMiddlingScore(self):
+    """When correct is half of the ceiling, the resulting score is 0.0."""
+    # To create the proposed file, clone the solution file and remove the
+    # virtual entity, which comprises 50% of the dictionary
+    proposed = copy.deepcopy(self.highest_score_argument[SOLUTION])
+    proposed.pop('daf3048f-2118-4afc-a5f0-8b4ce0c25fef')
```

Comment:
please create a file with the data you want, it will be cleaner to flush it out in a new file.

Reply:
I actually already have it — good catch.